lib/index.js

```diff
@@ -41,6 +41,10 @@ module.exports = function showdownHighlight({ pre = false } = {}) {
 
             const lang = (left.match(/class=\"([^ \"]+)/) || [])[1]
 
+            if (!lang && !auto_detection) {
+                return wholeMatch
+            }
+
```

Comment:
Early return if `auto_detection` is `false` and there is no language information. This prevents redundant operations.